### PR TITLE
Chart mouse gesture fix

### PR DESCRIPTION
**Pull request by robinschmid**
As requested by @SteffenHeu numerous times, a fix for the mouse chart gestures

### CODE REVIEW

**Comment by tomas-pluskal** [comment]
Thanks Robin!
Would it be possible to also place some limit on zooming out with the mouse gestures? Zooming out should not be possible beyond the actual range of the data.

File: src/main/java/io/github/mzmine/gui/chartbasics/gestures/ChartGestureHandler.java [hunk 1]
```diff
@@ -309,7 +310,9 @@ public static ChartGestureHandler createHandler(Handler handler, final ChartGest
           if (axis != null) {
             double diff = 0.03;
             if (e.getMouseEvent().isMouseWheelEvent()) {
-              diff = -0.10 * e.getMouseEvent().getWheelRotation();
+              // TODO actually get the mouse wheel distance and calculate percentage
+              diff = 0.10 * (e.getMouseEvent().getWheelRotation()>0? 1 : -1);
+              System.out.println("ZOOM "+e.getMouseEvent().getWheelRotation());
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/gui/chartbasics/gestures/ChartGestureHandler.java:315]: should this still be on?

File: src/main/java/io/github/mzmine/gui/chartbasics/gui/javafx/EChartViewer.java [hunk 2]
```diff
@@ -328,7 +327,7 @@ public void axisRangeChanged(ChartViewWrapper chart, ValueAxis axis, Range lastR
       if (standardGestures) {
         addStandardGestures();
       }
-      // mouseAdapter.addDebugHandler();
+      mouseAdapter.addDebugHandler();
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/gui/chartbasics/gui/javafx/EChartViewer.java:330]: should the debug handler still be enabled?

File: src/main/java/io/github/mzmine/gui/chartbasics/gui/javafx/EChartViewer.java [hunk 3]
```diff
@@ -492,6 +491,9 @@ public void setMouseZoomable(boolean flag) {
     setDomainZoomable(flag);
     setRangeZoomable(flag);
     isMouseZoomable = flag;
+    // TODO find better solution
+    // clear handler to stop zoom rectangle (hacky solution)
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/gui/chartbasics/gui/javafx/EChartViewer.java:495]: So you can't zoom by drawing a rect anymore? Why?

Comment (robinschmid) [reply src/main/java/io/github/mzmine/gui/chartbasics/gui/javafx/EChartViewer.java:495]: you can. This method is only triggered if you perform any mouse dragging over an entity that has a registered drag handler. 
So if we drag the mouse over the axes we zoom or scroll them and stop JFreeChart internal mouse handler - if we drag in the plot area we still zoom with the rect

File: src/main/java/io/github/mzmine/gui/chartbasics/gui/javafx/EChartViewer.java (hunk 2 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/gui/chartbasics/gui/javafx/EChartViewer.java:331]: Done

File: src/main/java/io/github/mzmine/gui/chartbasics/gestures/ChartGestureHandler.java (hunk 1 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/gui/chartbasics/gestures/ChartGestureHandler.java:315]: done

**Comment by tomas-pluskal** [comment]
Wonderful :)

**Comment by robinschmid** [comment]
Changed everything according to your comments.
Ready for merge from my side. 
The ranges are now limited by auto rage

**Comment by tomas-pluskal** [comment]
Thanks Robin!

**Pull request merged**